### ISSUE TITLE
Frontend generates invalid Bytes domain values when parsing from JSON to Kore

### ISSUE THREAD

**Issue by hjorthjort** [runtimeverification/k#1549]
In the Polkadot Semantics, we run this file:

module.wat
```
(module (memory 0))
```

Then run
```
krun --directory .build/defn/kwasm/haskell  module.wat --output json > module.loaded.json
```

Producing the following JSON:

module.loaded.json
```
{"format":"KAST","version":1,"term":{"node":"KApply","label":"<generatedTop>","variable":false,"arity":2,"args":[{"node":"KApply","label":"<polkadot-host>","variable":false,"arity":3,"args":[{"node":"KApply","label":"<wasm>","variable":false,"arity":9,"args":[{"node":"KApply","label":"<instrs>","variable":false,"arity":1,"args":[{"node":"KSequence","arity":0,"items":[]}]},{"node":"KApply","label":"<valstack>","variable":false,"arity":1,"args":[{"node":"KApply","label":".ValStack_WASM-DATA_ValStack","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<curFrame>","variable":false,"arity":2,"args":[{"node":"KApply","label":"<locals>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<curModIdx>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]}]},{"node":"KApply","label":"<moduleRegistry>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<moduleIds>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<moduleInstances>","variable":false,"arity":1,"args":[{"node":"KApply","label":"ModuleInstCellMapItem","variable":false,"arity":2,"args":[{"node":"KApply","label":"<modIdx>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]},{"node":"KApply","label":"<moduleInst>","variable":false,"arity":14,"args":[{"node":"KApply","label":"<modIdx>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]},{"node":"KApply","label":"<exports>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<types>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<nextTypeIdx>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]},{"node":"KApply","label":"<funcAddrs>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<nextFuncIdx>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]},{"node":"KApply","label":"<tabIds>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<tabAddrs>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<memIds>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<memAddrs>","variable":false,"arity":1,"args":[{"node":"KApply","label":"_|->_","variable":false,"arity":2,"args":[{"node":"KToken","sort":"Int","token":"0"},{"node":"KToken","sort":"Int","token":"0"}]}]},{"node":"KApply","label":"<globIds>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<globalAddrs>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<nextGlobIdx>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]},{"node":"KApply","label":"<moduleMetadata>","variable":false,"arity":3,"args":[{"node":"KApply","label":"<moduleId>","variable":false,"arity":1,"args":[{"node":"KApply","label":"_WASM-DATA_OptionalId","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<funcIds>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<typeIds>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Map","variable":false,"arity":0,"args":[]}]}]}]}]}]},{"node":"KApply","label":"<nextModuleIdx>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"1"}]},{"node":"KApply","label":"<mainStore>","variable":false,"arity":8,"args":[{"node":"KApply","label":"<funcs>","variable":false,"arity":1,"args":[{"node":"KApply","label":".FuncDefCellMap","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<nextFuncAddr>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]},{"node":"KApply","label":"<tabs>","variable":false,"arity":1,"args":[{"node":"KApply","label":".TabInstCellMap","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<nextTabAddr>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]},{"node":"KApply","label":"<mems>","variable":false,"arity":1,"args":[{"node":"KApply","label":"MemInstCellMapItem","variable":false,"arity":2,"args":[{"node":"KApply","label":"<mAddr>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]},{"node":"KApply","label":"<memInst>","variable":false,"arity":4,"args":[{"node":"KApply","label":"<mAddr>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]},{"node":"KApply","label":"<mmax>","variable":false,"arity":1,"args":[{"node":"KApply","label":".Int_WASM-DATA_OptionalInt","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<msize>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]},{"node":"KApply","label":"<mdata>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Bytes","token":"b\"\""}]}]}]}]},{"node":"KApply","label":"<nextMemAddr>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"1"}]},{"node":"KApply","label":"<globals>","variable":false,"arity":1,"args":[{"node":"KApply","label":".GlobalInstCellMap","variable":false,"arity":0,"args":[]}]},{"node":"KApply","label":"<nextGlobAddr>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]}]},{"node":"KApply","label":"<deterministicMemoryGrowth>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Bool","token":"true"}]}]},{"node":"KApply","label":"<k>","variable":false,"arity":1,"args":[{"node":"KSequence","arity":0,"items":[]}]},{"node":"KApply","label":"<trace>","variable":false,"arity":1,"args":[{"node":"KApply","label":".List","variable":false,"arity":0,"args":[]}]}]},{"node":"KApply","label":"<generatedCounter>","variable":false,"arity":1,"args":[{"node":"KToken","sort":"Int","token":"0"}]}]}}
```

We try to run it with `krun --directory .build/defn/kwasm/haskell --output json --parser cat  --term module.loaded.json`

Output:
```
kore-exec: [1607414] Error (ErrorException):
    Error:
      symbol or alias 'Lbl'-LT-'generatedTop'-GT-'' (/home/hjorthjort/polkadot-verification/dev/./.krun-2020-09-18-13-11-17-449-8cca0783-a410-4d7b-9430-cf0c6809a1fc/execute-initial.kore 1:1):
      symbol or alias 'Lbl'-LT-'polkadot-host'-GT-'' (/home/hjorthjort/polkadot-verification/dev/./.krun-2020-09-18-13-11-17-449-8cca0783-a410-4d7b-9430-cf0c6809a1fc/execute-initial.kore 1:31):
      symbol or alias 'Lbl'-LT-'wasm'-GT-'' (/home/hjorthjort/polkadot-verification/dev/./.krun-2020-09-18-13-11-17-449-8cca0783-a410-4d7b-9430-cf0c6809a1fc/execute-initial.kore 1:62):
      symbol or alias 'Lbl'-LT-'mainStore'-GT-'' (/home/hjorthjort/polkadot-verification/dev/./.krun-2020-09-18-13-11-17-449-8cca0783-a410-4d7b-9430-cf0c6809a1fc/execute-initial.kore 1:1511):
      symbol or alias 'Lbl'-LT-'mems'-GT-'' (/home/hjorthjort/polkadot-verification/dev/./.krun-2020-09-18-13-11-17-449-8cca0783-a410-4d7b-9430-cf0c6809a1fc/execute-initial.kore 1:1742):
      symbol or alias 'LblMemInstCellMapItem' (/home/hjorthjort/polkadot-verification/dev/./.krun-2020-09-18-13-11-17-449-8cca0783-a410-4d7b-9430-cf0c6809a1fc/execute-initial.kore 1:1764):
      symbol or alias 'Lbl'-LT-'memInst'-GT-'' (/home/hjorthjort/polkadot-verification/dev/./.krun-2020-09-18-13-11-17-449-8cca0783-a410-4d7b-9430-cf0c6809a1fc/execute-initial.kore 1:1832):
      symbol or alias 'Lbl'-LT-'mdata'-GT-'' (/home/hjorthjort/polkadot-verification/dev/./.krun-2020-09-18-13-11-17-449-8cca0783-a410-4d7b-9430-cf0c6809a1fc/execute-initial.kore 1:2017):
      \dv (/home/hjorthjort/polkadot-verification/dev/./.krun-2020-09-18-13-11-17-449-8cca0783-a410-4d7b-9430-cf0c6809a1fc/execute-initial.kore 1:2044):
      Verifying builtin sort 'BYTES.Bytes':
      While parsing domain value:
        <string literal>:1:2:
          |
        1 | b""
          |  ^
        unexpected '"'
        expecting hex digit
        
    CallStack (from HasCallStack):
      error, called at app/share/GlobalMain.hs:438:13 in main:GlobalMain
Created bug report: kore-exec.tar.gz
```
[kore-exec.tar.gz](https://github.com/kframework/kore/files/5245717/kore-exec.tar.gz)

Attaching the generated bugreport.

**Comment by dwightguth** [comment]
The code that converts from K to kore is supposed to demangle this representation, which is a correct K representation but not a correct KORE representation. Looks like it's not happening.

**Comment by ehildenb** [comment]
@hjorthjort we are having problems understanding what's causing this problem. I remember that initially we found this by trying to feed the same input into the haskell bakcend that we fed into the llvm backend (without problems), and that was only using `kast` to produce the resulting JSON that was needed.

In this example you are calling `krun` instead, which exercises different code-paths. So we are concerned that the fix that Radu made actually won't fix the primary cause, because it doesn't make sense that this function is called when you call `kast`.

Please post the original problem and the workflow to reproduce it as well.

**Comment by radumereuta** [comment]
Fixed by #1692
